extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/db/updater/CoordinatorBasicAuthorizerMetadataStorageUpdater.java

```diff
@@ -0,0 +1,761 @@
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import com.metamx.emitter.EmittingLogger;
+import io.druid.common.config.ConfigManager;
+import io.druid.concurrent.LifecycleLock;
+import io.druid.guice.ManageLifecycle;
+import io.druid.guice.annotations.Smile;
+import io.druid.java.util.common.ISE;
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.concurrent.Execs;
+import io.druid.java.util.common.concurrent.ScheduledExecutors;
+import io.druid.java.util.common.lifecycle.LifecycleStart;
+import io.druid.java.util.common.lifecycle.LifecycleStop;
+import io.druid.metadata.MetadataCASUpdate;
+import io.druid.metadata.MetadataStorageConnector;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.security.basic.BasicSecurityDBResourceException;
+import io.druid.security.basic.BasicAuthCommonCacheConfig;
+import io.druid.security.basic.authorization.BasicRoleBasedAuthorizer;
+import io.druid.security.basic.authorization.db.cache.BasicAuthorizerCacheNotifier;
+import io.druid.security.basic.authorization.entity.BasicAuthorizerPermission;
+import io.druid.security.basic.authorization.entity.BasicAuthorizerRole;
+import io.druid.security.basic.authorization.entity.BasicAuthorizerUser;
+import io.druid.security.basic.authorization.entity.UserAndRoleMap;
+import io.druid.server.security.Action;
+import io.druid.server.security.Authorizer;
+import io.druid.server.security.AuthorizerMapper;
+import io.druid.server.security.Resource;
+import io.druid.server.security.ResourceAction;
+import io.druid.server.security.ResourceType;
+import org.joda.time.Duration;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@ManageLifecycle
+public class CoordinatorBasicAuthorizerMetadataStorageUpdater implements BasicAuthorizerMetadataStorageUpdater
+{
+  private static final EmittingLogger LOG =
+      new EmittingLogger(CoordinatorBasicAuthorizerMetadataStorageUpdater.class);
+
+  public static final List<ResourceAction> SUPERUSER_PERMISSIONS = makeSuperUserPermissions();
+
+  public static final String USERS = "users";
+  public static final String ROLES = "roles";
+  public static final TypeReference USER_MAP_TYPE_REFERENCE = new TypeReference<Map<String, BasicAuthorizerUser>>()
+  {
+  };
+  public static final TypeReference ROLE_MAP_TYPE_REFERENCE = new TypeReference<Map<String, BasicAuthorizerRole>>()
+  {
+  };
+  public static final TypeReference USER_AND_ROLE_MAP_TYPE_REFERENCE = new TypeReference<UserAndRoleMap>()
```

Comment:
Suggest to move the above TypeReferences to some util class to remove duplicated codes in ```CoordinatorBasicAuthenticatorMetadataStorageUpdater```.

Reply:
Moved type references to BasicAuthUtil